### PR TITLE
Add change log for v10.0.0-rc.5 release.

### PR DESCRIPTION
**Pull request by pengdev**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.

### Summary of changes

Add change log for v10.0.0-rc.5 release.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,24 @@
 
 Mapbox welcomes participation and contributions from everyone.
 
+# 10.0.0-rc.5 July 28, 2021
+
+**The Mapbox Maps SDK for Android has moved to release candidate status and is now ready for production use.**
+
+## Breaking changes ⚠️
+* Improve camera API consumption from java programming language by adding a JvmName annotations. ([#495](https://github.com/mapbox/mapbox-maps-android/pull/495))
```
Comment (pengdev) [inline CHANGELOG.md:10]: Does this PR only changes the Java signature? So it's a breaking change only for Java users?

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,24 @@
...
+
+## Features ✨ and improvements 🏁
+* Add showLogo and showAttributes config for snapshotter. User can then hide logo and attributions in a snapshotter ([#496](https://github.com/mapbox/mapbox-maps-android/pull/496))
+* Add lifecycle plugin, users don't need to call onStart/onStop/onDestroy methods with this plugin ([#485](https://github.com/mapbox/mapbox-maps-android/pull/485))
+* Add a minimum Android Auto test app and an optional Android Auto extension that provide convenient extension function to initialise the MapSurface from a Car App Session. ([#488](https://github.com/mapbox/mapbox-maps-android/pull/488))
+* Add lint check for lifecycle methods ([#516](https://github.com/mapbox/mapbox-maps-android/pull/516))
+* Get rid of reflection for creating plugins. That will lead to decreasing MapView startup time if plugins are enabled. ([#519](https://github.com/mapbox/mapbox-maps-android/pull/519))
```
Comment (kiryldz) [inline CHANGELOG.md:18]: that's a breaking change, more info in PR description

File: CHANGELOG.md (hunk 1 above)
Comment (kiryldz) [inline CHANGELOG.md:10]: Can we be a bit more specific here? or should we be more specific in release notes?

File: CHANGELOG.md (hunk 2 above)
Comment (pengdev) [reply CHANGELOG.md:18]: seems it is also marked as feature label, guess that's why it ended up in this section

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,24 @@
...
+
+**The Mapbox Maps SDK for Android has moved to release candidate status and is now ready for production use.**
+
+## Breaking changes ⚠️
+* Improve camera API consumption from java programming language by adding a JvmName annotations. ([#495](https://github.com/mapbox/mapbox-maps-android/pull/495))
+* Rename AttributionView#setOnClickListener to setViewOnClickListener to avoid overloading the Android SDK method. Results in compilation on Android P and above. Adjust codebase to changes in enforced nullability of Android SDK code. ([#497](https://github.com/mapbox/mapbox-maps-android/pull/497))
+
+## Features ✨ and improvements 🏁
+* Add showLogo and showAttributes config for snapshotter. User can then hide logo and attributions in a snapshotter ([#496](https://github.com/mapbox/mapbox-maps-android/pull/496))
+* Add lifecycle plugin, users don't need to call onStart/onStop/onDestroy methods with this plugin ([#485](https://github.com/mapbox/mapbox-maps-android/pull/485))
```
Comment (kiryldz) [inline CHANGELOG.md:15]:
```suggestion
* Add lifecycle plugin so there is no need to call onStart/onStop/onDestroy methods explicitly ([#485](https://github.com/mapbox/mapbox-maps-android/pull/485))
```

File: CHANGELOG.md (hunk 1 above)
Comment (pengdev) [reply CHANGELOG.md:10]: I think we should be both specific and concise here, and maybe we can expand further in the release notes.

File: CHANGELOG.md (hunk 2 above)
Comment (pengdev) [reply CHANGELOG.md:18]: Also, maybe we can add more specifics here about the exact API breaking change.

File: CHANGELOG.md (hunk 3 above)
Comment (pengdev) [reply CHANGELOG.md:15]: I guess we should maybe mention the limitations as well, wdyt?

File: CHANGELOG.md (hunk 2 above)
Comment (kiryldz) [reply CHANGELOG.md:18]: I've marked it with both labels, my fault, script seems to respect only the first label.

File: CHANGELOG.md (hunk 1 above)
Comment (kiryldz) [reply CHANGELOG.md:10]: Yes, it should be relevant for Java only users.

File: CHANGELOG.md (hunk 2 above)
Comment (pengdev) [reply CHANGELOG.md]: no worries, we can fix it afterwards anyway 😄

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,24 @@
...
 Mapbox welcomes participation and contributions from everyone.
 
+# 10.0.0-rc.5 July 28, 2021
+
+**The Mapbox Maps SDK for Android has moved to release candidate status and is now ready for production use.**
+
+## Breaking changes ⚠️
+* Improve camera API consumption from java programming language by adding `CameraAnimationsUtils` and `getCamera` JvmName annotations. ([#495](https://github.com/mapbox/mapbox-maps-android/pull/495))
+* Rename `AttributionView#setOnClickListener` to `setViewOnClickListener` to avoid overloading the Android SDK method. Results in compilation on Android P and above. Adjust codebase to changes in enforced nullability of Android SDK code. ([#497](https://github.com/mapbox/mapbox-maps-android/pull/497))
+* Get rid of using reflection when creating plugins. Which will decrease MapView startup time if plugins are enabled. ([#519](https://github.com/mapbox/mapbox-maps-android/pull/519))
```
Comment (kiryldz) [inline CHANGELOG.md:12]:
```suggestion
* Get rid of using reflection when creating plugins which should decrease `MapView` startup time if plugins are enabled. ([#519](https://github.com/mapbox/mapbox-maps-android/pull/519))
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,24 @@
...
+
+## Features ✨ and improvements 🏁
+* Add `showLogo` and `showAttributes` config for snapshotter. User can then hide logo and attributions in a snapshotter ([#496](https://github.com/mapbox/mapbox-maps-android/pull/496))
```
Comment (kiryldz) [inline CHANGELOG.md:15]:
```suggestion
* Add `showLogo` and `showAttributes` config for snapshotter which are defaulted to true. User can then hide logo and attributions in a snapshotter ([#496](https://github.com/mapbox/mapbox-maps-android/pull/496))
```

**Review by kiryldz — APPROVED** [review]
LGTM guess we could merge it without waiting for CI to finish

**Pull request merged**